### DISCUSSION TITLE
UK Tesla Powerwall 3 + Gateway 2 - Powerwall 3 mode fails on Access Denied: Check your Gateway Password

### DISCUSSION THREAD

**Discussion by PASGT** [jasonacox/Powerwall-Dashboard#551] [General]
Hi,
I have a Powerwall 3 and Gateway 2.

I have got this working via the Tesla Cloud (Option 2) but I cannot get it working via Powerwall 3 (Option 4).

I have the Gateway password from the inside of the Gateway, but it keeps saying as below "Access Denied: Check your Gateway Password"
The password is the one on the sticker. It also has a QR Code which when scanned gives the information below the log

Any help would be most welcome.

`12/06/2024 04:27:55 PM [proxy] [INFO] pyPowerwall [0.12.0] Proxy Server [t66] - HTTP Port 8675`
`12/06/2024 04:27:55 PM [proxy] [INFO] pyPowerwall Proxy Started`
`12/06/2024 04:27:55 PM [pypowerwall.tedapi] [ERROR] Access Denied: Check your Gateway Password`
`12/06/2024 04:27:55 PM [pypowerwall.tedapi] [ERROR] Failed to connect to Powerwall Gateway`
`12/06/2024 04:27:55 PM [pypowerwall.tedapi] [ERROR] Access Denied: Check your Gateway Password`
`12/06/2024 04:27:55 PM [pypowerwall.fleetapi.pypowerwall_fleetapi] [ERROR] Failed to retrieve sitelist - MissingSchema("Invalid URL /api/1/products': No scheme supplied. Perhaps you meant https:///api/1/products?")'
`12/06/2024 04:27:55 PM [pypowerwall.fleetapi.pypowerwall_fleetapi] [ERROR] No sites found for`
`12/06/2024 04:27:55 PM [pypowerwall] [ERROR] Failed to connect to Powerwall with all modes. Waiting 30s to retry.`

QR Code Scan of the Gateway
`WIFI:T:WPA;S:TEG-00Z;P:JLDGCSGGZU;;P:1153100-13-L:S:CN322216G3L00Z`

Outputs from verify.sh
Verify Powerwall-Dashboard 4.6.0 on Linux - Timezone: Europe/London
----------------------------------------------------------------------------
This script will attempt to verify all the services needed to run
Powerwall-Dashboard. Use this output when you open an issue for help:
https://github.com/jasonacox/Powerwall-Dashboard/issues/new

Checking pypowerwall
----------------------------------------------------------------------------
 - Config File pypowerwall.env: GOOD
 - Container (pypowerwall): GOOD
 - **Service (port 8675): ERROR: Not Listening**
 - Version: Unknown
 - Powerwall State: **ERROR: Not Connected - Firmware: Unknown**
 - Gateway TEDAPI: Available (192.168.91.1)
 - TEDAPI Vitals: **Not Connected**
 - Cloud Mode: NO

Checking telegraf
----------------------------------------------------------------------------
 - Config File telegraf.conf: GOOD
 - Local Config File telegraf.local: GOOD
 - Container (telegraf): GOOD
 - Version: Telegraf 1.28.2 (git: HEAD@8d9cf395)

Checking influxdb
----------------------------------------------------------------------------
 - Config File influxdb.conf: GOOD
 - Environment File influxdb.env: GOOD
 - Container (influxdb): GOOD
 - Service (port 8086): GOOD
 - Filesystem (./influxdb): GOOD
 - Version: InfluxDB shell version: 1.8.10

Checking grafana
----------------------------------------------------------------------------
 - Config File grafana.env: GOOD
 - Container (grafana): GOOD
 - Service (port 9000): GOOD
 - Filesystem (./grafana): GOOD
 - Version: Grafana CLI version 9.1.2

Checking weather411
----------------------------------------------------------------------------
 - Container (weather411): GOOD
 - Service (port 8676): GOOD
 - Weather: {"temperature": 4.2}
 - Version: 0.2.3

One or more tests failed.

**Comment by PASGT** [comment]
OK - Update - Got it working...

In the UK at least it is NOT the password that is on the Gateway.
You need to use the password that is on the actual Powerwall 3. 

I also thought these were the WIFI passwords rather than API, but anyway. Sorted now - I was using the Gateway password when it needs the Powerwall password. 

Hopefully this helps someone else.

**Reply by jasonacox** [reply]
Thanks @PASGT - I'm wondering what instructions we could add to README, perhaps in the Troubleshooting section, that would help PW3 owners. Where did you find the password that worked?

**Reply by olsw** [reply]
Spent a while working this out too! There are two passwords, one on the gateway sticker and one on the powerwall sticker. As above the one that is required is the PowerWall password and *NOT* the gateway one. Spent far too long trying to understand what was going wrong!
Is the US different then?

**Reply by AG7GN** [reply]
I just installed this configuration (Powerwall 3 + Gateway 2) here in the U.S. and I found the same: You must use the PW3's password, not the gateway's password. Also, the ethernet cable I ran to the gateway is not used - the ethernet port doesn't even turn on when the gateway is connected to the PW3. Instead, I've set the PW3 to connect to my WiFi for network access. I should've ran the ethernet cable to the PW3 instead, but live and learn.

**Reply by jasonacox** [reply]
That's good to know! That is not the case with the PW2 and PW+ system I have. Hum. I'm thinking I should add a note:

* Powerwall 3 Owners - The password you will use for local TEDAPI setup is the one on the sticker on the PW3 itself, not the one on the Gateway.

And then edit the setup.sh script ... which is currently: 

```bash
                echo ""
                echo "Congratulations!"
                echo "Extended Device Metrics (vitals) are available on this endpoint via TEDAPI."
                echo "However, you will need the Gateway password to access them."
                echo "This password is often on the QR code on the Powerwall Gateway unit."
                echo ""
                read -p 'Enter Gateway Password or leave blank to disable: ' PW
```

**Reply by jasonacox** [reply]
Minor update that may help: https://github.com/jasonacox/Powerwall-Dashboard/commit/f5eba98cb01548eb4953984a9ca35b04da450d7f

**Reply by AG7GN** [reply]
Thanks, Jason. This change is helpful. It would also help if elsewhere in the documentation to explain that for those with PW3 +GW2 wanting to use an ethernet cable rather than WiFi to collect data, be sure to run the cable to the PW3 and not the GW2.

**Reply by jasonacox** [reply]
Good call - Added - https://github.com/jasonacox/Powerwall-Dashboard/commit/6a717897521933da53775bd936399099f893e909
